boot/freeldr/freeldr/arch/realmode/helpers_pc98.inc

```diff
@@ -80,27 +80,39 @@ writechr:
     pushf
     pusha
 
+    /* Check if the VRAM segment was initialized */
+    mov dx, word ptr ds:[VramSegment]
+    test dx, dx
+    jnz short .writechr_write
+
     /* High-resolution mode check */
     test byte ptr ds:[HEX(501)], HEX(08)
-    jz .writechr_normal
-    push HEX(0E000)
-    jmp short .writechr_test_done
-.writechr_normal:
-    push HEX(0A000)
-
+    mov dx, HEX(0A000) // Suppose normal mode
+    jz short .writechr_test_done
+    mov dh, HEX(E0)    // Change 0xA000 to 0xE000, use hi-res mode
```

Comment:
@disean Note that here I fixed the value to be E0, instead of 0E as in your proposal, because it had to change the A0 to E0.